tests/test_model.py

```diff
@@ -168,3 +168,25 @@ def test_crop_model_load_checkpoint_with_explicit_num_classes(tmpdir, crop_model
     # Check model parameters match
     for p1, p2 in zip(crop_model.parameters(), loaded_model.parameters()):
         assert torch.equal(p1, p2)
+
+
+def test_expand_bbox_to_square_edge_cases():
+    """Test cases for the expand_bbox_to_square function."""
+    crop_model = model.CropModel(num_classes=2)
+    
+    # Test Case 1: Bounding box at the image edge (0,0)
+    bbox = [0, 0, 20, 30]
+    image_width, image_height = 100, 100
+    # Expected to expand vertically to make a square while remaining within image bounds
+    expected = [0.0, 0.0, 30.0, 30.0]
+    result = crop_model.expand_bbox_to_square(bbox, image_width, image_height)
+    assert result == expected
+
+    # Test Case 2: Side length exceeds both image dimensions
+    bbox = [10, 10, 180, 180] 
+    image_width, image_height = 100, 100
+    # Expected to be clamped to the size of the image: full image bounding box
+    expected = [0.0, 0.0, 100.0, 100.0]
+    result = crop_model.expand_bbox_to_square(bbox, image_width, image_height)
+    assert result == expected
+
```

Comment:
This is great. Can we add one more basic test where there are no edges that influence the expansion